packages/perseus/src/renderer.tsx

```diff
@@ -1706,7 +1700,24 @@ class Renderer extends React.Component<Props, State> {
                 return widget.getUserInput();
             }
         });
-    };
+    }
+
+    /**
+     * Returns an object of the widget `.getUserInput()` results
+     */
+    getUserInputMap(): UserInputMap {
+        const userInputMap = {};
+        this.widgetIds.forEach((id: string) => {
+            const widget = this.getWidgetInstance(id);
+            // Handle Groups, which have their own sets of widgets
+            if (widget?.getUserInputMap) {
+                userInputMap[id] = widget.getUserInputMap();
+            } else if (widget?.getUserInput) {
+                userInputMap[id] = widget.getUserInput();
+            }
```

Comment:
Why does Group need to be different from the other widgets here? Is the issue that we can't change the interface of `getUserInput` because it needs to stay compatible with e.g. Khanmigo?

Reply:
IIRC, I couldn't change the format of `getUserInput` while `guessAndScore` exists. The next step is to break `guessAndScore` though, so I filed a ticket to followup on this once that's done: https://khanacademy.atlassian.net/browse/LEMS-2523